plugins/dell-kestrel/fu-dell-kestrel-ec.c

```diff
@@ -76,13 +76,15 @@ fu_dell_kestrel_ec_is_dev_present(FuDellKestrelEc *self,
 gboolean
 fu_dell_kestrel_ec_is_chunk_supported(FuDevice *device, FuDellKestrelEcDevType dev_type)
 {
-	FuDellKestrelEc *self = FU_DELL_KESTREL_EC(device);
+	FuDellKestrelEc *self = NULL;
 	guint8 chunk_support = 0;
 
 	switch (dev_type) {
 	case FU_DELL_KESTREL_EC_DEV_TYPE_RMM:
 		return FALSE;
 	case FU_DELL_KESTREL_EC_DEV_TYPE_PD:
+		/* casting to EC based on the device type */
+		self = FU_DELL_KESTREL_EC(device);
```

Comment:
this isn't really idiomatic -- self is supposed to be the same GType as the prefix - i.e. fu_dell_kestrel_ec() -- what GType is being passed to this?

Reply:
the device could be a `fu_dell_kestrel_rmm`, if that is the case we can't cast it to `fu_dell_kestrel_ec`. So need to know the `dev_type` first, e.g.: if `dev_type` is `pd` then the device here is essentially pd's `proxy` in terms of `ec` and the casting is surely expected.